File: modules/coupon-domain/src/main/java/com/huythanh0x/udemycoupons/dto/ReviewsSummaryDTO.java
package com.huythanh0x.udemycoupons.dto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

/**
 * DTO for reviews summary information.
 */
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class ReviewsSummaryDTO {
    private Integer totalCount;
    private Float averageRating;
    private List<ReviewDTO> recentReviews; // First 3-5 reviews
}

File: modules/coupon-domain/src/main/kotlin/com/thanh0x/coursedeal/dto/ReviewsSummaryDTO.kt
package com.thanh0x.coursedeal.dto

/**
 * DTO for reviews summary information.
 */
data class ReviewsSummaryDTO(
    val totalCount: Int? = null,
    val averageRating: Float? = null,
    // First 3-5 reviews
    val recentReviews: List<ReviewDTO>? = null,
)
